Notes/Features/SignIn/ContentViewModel.swift

```diff
@@ -29,23 +35,55 @@ final class ContentViewModel: ObservableObject {
 	func onScreenEvent(_ event: ScreenEvent) {
 		switch event {
 			case .signIn:
+				compairUserInputAndGeneratedCode()
+			case .generateCode:
 				valiateInput()
+				if shouldCodeBeSended == true {
+					self.validCode = self.generateCodeNumber()
```

Comment:
we only write `self.` when it's needed:
- to distinguish between class/struct variable with the local variable from the current context
- in some block, like with Task, ...